### DISCUSSION TITLE
openai vision not working after update librechat

### DISCUSSION THREAD

**Discussion by assassinliujie** [danny-avila/LibreChat#2303] [Troubleshooting] [answered]
### What happened?

I update my librechat after 3month, and the gpt vision stop working

### Steps to Reproduce

1.git pull 2.docker-compose pull 

### What browsers are you seeing the problem on?

Microsoft Edge

### Relevant log output

```shell
LibreChat         | 2024-04-03 19:45:18 warn: [OpenAIClient.chatCompletion][stream] API error
LibreChat         | 2024-04-03 19:45:18 error: [handleAbortError] AI response error; aborting request: 400  (request id: 2024040403451755785221215164114)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by assassinliujie** [comment]
![f5b45b239b07e12b8d252ebe416ebbec](https://github.com/danny-avila/LibreChat/assets/68693675/73e9ece1-5a4a-48ab-ace0-dc1350257a1f)

**Comment by assassinliujie** [comment]
It says image_url not supportted, maybe we should try the base64 mode but I don't know how to

**Comment by assassinliujie** [comment]
the RAG is also not working. It keeps uploading file but not even sending a request to the openai embedding

**Comment by danny-avila** [comment]
It is working. You either have a configuration issue or you did not update correctly. I’m inclined to think the latter. How did you update exactly?

**Reply by danny-avila** [reply]
Check the `git log`

**Reply by assassinliujie** [reply]
I am trying a fresh installation on another server without going through any reverse proxies, as I suspect there might be some issue with the reverse proxy. My update method is to git clone a new project to a fresh location and then move the data node and env file.

**Reply by assassinliujie** [reply]
LibreChat         | 2024-04-04 06:52:37 warn: [OpenAIClient.chatCompletion][stream] API error
LibreChat         | 2024-04-04 06:52:37 error: [handleAbortError] AI response error; aborting request: 400 Invalid content type. image_url is only supported by certain models.
I have already tried reinstalling and I am not using any reverse proxies, but errors still occur. Other text models work normally. My configuration file is as follows.
![image](https://github.com/danny-avila/LibreChat/assets/68693675/d381c5d5-7146-49b5-ad46-5bbe519e2ccc)

**Reply by assassinliujie** [reply, marked as answer]
Hello, I have figured out the problem. I need to include gpt-4-vision-preview and text-embedding-3-small in the available models to operate normally. I remember it was not necessary before, at least, vision was not needed.
![image](https://github.com/danny-avila/LibreChat/assets/68693675/ad2cbc75-fc92-4415-868a-45c83c54cebe)

**Reply by assassinliujie** [reply]
![image](https://github.com/danny-avila/LibreChat/assets/68693675/14b0cb7d-9870-4c0f-8866-77b971ee14e9)
but this will cause the vision model to display in the wrong location.

**Comment by danny-avila** [comment]
On the latest commit, I can even use vision + RAG at the same time:

![image](https://github.com/danny-avila/LibreChat/assets/110412045/ea80e022-d89b-4760-9c41-c79fa8c17cc3)

**Comment by danny-avila** [comment]
> Hello, I have figured out the problem. I need to include gpt-4-vision-preview and text-embedding-3-small in the available models to operate normally. I remember it was not necessary before, at least, vision was not needed.

`text-embedding-3-small` should not be necessary but yes, excellent debugging. Apologies, I forgot this. It was introduced for better security if the admin did not make vision available.

**Reply by danny-avila** [reply]
vision can be used as a standalone model, it's not necessarily in the wrong location
